File: tools/test/py_harness/base/spider.py
"""影视TV chaquo 模块里 base/spider.py 的等价物（仅用于沙盒加载测试）"""


class Spider:
    def getName(self):
        return ""

    def init(self, extend=""):
        pass

    def homeContent(self, filter):
        return {}

    def homeVideoContent(self):
        return {}

    def categoryContent(self, tid, pg, filter, extend):
        return {}

    def detailContent(self, ids):
        return {}

    def searchContent(self, key, quick, pg="1"):
        return {}

    def playerContent(self, flag, id, vipFlags):
        return {}

    def liveContent(self, url):
        return {}

    def isVideoFormat(self, url):
        return False

    def manualVideoCheck(self):
        return False

    def destroy(self):
        pass

    def localProxy(self, param):
        return None

File: src/main/java/com/github/catvod/crawler/Spider.java
package com.github.catvod.crawler;

import android.content.Context;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Stub of catvod's Spider class — used at SDK build time.
 * Real Spider from catvod requires okhttp. At runtime, the host shell provides
 * the real one with identical method signatures.
 *
 * The compiled Spider.class is excluded from the final jar (see build.sh).
 */
public abstract class Spider {
    public String siteKey;

    public void init(Context context) throws Exception {}
    public void init(Context context, String extend) throws Exception { init(context); }
    public String homeContent(boolean filter) throws Exception { return ""; }
    public String homeVideoContent() throws Exception { return ""; }
    public String categoryContent(String tid, String pg, boolean filter, HashMap<String, String> extend) throws Exception { return ""; }
    public String detailContent(List<String> ids) throws Exception { return ""; }
    public String searchContent(String key, boolean quick) throws Exception { return ""; }
    public String searchContent(String key, boolean quick, String pg) throws Exception { return ""; }
    public String playerContent(String flag, String id, List<String> vipFlags) throws Exception { return ""; }
    public String liveContent(String url) throws Exception { return ""; }
    public boolean manualVideoCheck() throws Exception { return false; }
    public boolean isVideoFormat(String url) throws Exception { return false; }
    public Object[] proxy(Map<String, String> params) throws Exception { return null; }
    public String action(String action) throws Exception { return null; }
    public void destroy() {}
}
